src/LightInject.Interception/LightInject.Interception.cs

```diff
@@ -1807,8 +1807,17 @@ private MethodBuilder ImplementInterceptedMethod(MethodInfo targetMethod, int[]
                 
             if (proxyDefinition.TargetType.GetTypeInfo().IsInterface && proxyDefinition.ImplementingType != null)
             {
-                targetInvocationMethod = proxyDefinition.ImplementingType.GetTypeInfo().DeclaredMethods.FirstOrDefault(m => m.Name == targetMethod.Name);
-            }            
+                var type = proxyDefinition.ImplementingType;
+
+                while (type != null)
```

Comment:
When targeting frameworks **other than** .NET Standard 1.1 (e.g. .NET Standard 2.0, .NET Framework 4.5), we can use `Type.GetInterfaceMap`, which matches methods properly even when there are methods with the same name.